src/codegen.cpp

```diff
@@ -6952,6 +6954,20 @@ static std::pair<std::unique_ptr<Module>, jl_llvm_functions_t>
         JL_UNLOCK(&m->writelock);
     }
 
+    // link the dependent llvmcall modules, but switch their function's linkage to private
+    // so that they don't show up in the execution engine.
+    for (auto &Mod : ctx.llvmcall_modules) {
+        SmallVector<std::string, 1> Exports;
+        for (const auto &F: Mod->functions())
+            if (!F.isDeclaration())
+                Exports.push_back(F.getName());
+        if (Linker::linkModules(*jl_Module, std::move(Mod))) {
+            jl_error("Failed to link LLVM bitcode");
+        }
+        for (auto FN: Exports)
+            jl_Module->getFunction(FN)->setLinkage(GlobalVariable::PrivateLinkage);
```

Comment:
I couldn't figure out a way to do this more cleanly. Using `PrivateLinkage` when creating the function in `emit_llvmcall` causes it not to get linked, just keeping the `LinkOnceODRLinkage` causes our code to [look-up the llvmcall function symbol](https://github.com/JuliaLang/julia/blob/0e74b3e872f6133704c4aed51fa99a6959ad17b1/src/jitlayers.cpp#L462-L463) and fail, while setting the symbol visibility to `HiddenVisibility` makes ORC throw an undefined symbol error when in `RuntimeDyldImpl::resolveRelocations`. @vtjnash @Keno thoughts?

We didn't run into this because we used to always set `alwaysinline`, but with this whole-module API it seems better to me to give the caller control about inlining, and only `alwaysinline` when passing function IR.